File: CodingInterviewsJavaSolutions/215. Kth Largest Element in an Array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        int sz = nums.length;

        for (int i = 0; i < sz; i++) {
            if (minHeap.size() <= k || nums[i] > minHeap.peek()) {
                minHeap.offer(nums[i]);
            }

            if (minHeap.size() > k) {
                minHeap.poll();
            }
        }

        return minHeap.peek();
    }
}

File: CodingInterviewsPackSolutionsCPP/215. Kth Largest Element in an Array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int,vector<int>,greater<int>>minHeap;
        int sz = nums.size();

        for(int i = 0; i < sz; i++){
            if(minHeap.size()<=k||nums[i]>minHeap.top()){
                minHeap.push(nums[i]);
            }

            if(minHeap.size()>k)
                minHeap.pop();
        }
        return minHeap.top();
    }
};